src/csharp/Grpc.Core/CallOptions.cs

```diff
@@ -236,22 +236,24 @@ internal CallOptions WithFlags(CallFlags flags)
         internal CallOptions Normalize()
         {
             var newOptions = this;
+            // silently ignore the context propagation token if it wasn't produced by "us"
+            var propagationTokenImpl = propagationToken.AsImplOrNull();
```

Comment:
should this null check be on `propagationTokenImpl` rather than `propagationToken` ?

Reply:
Good catch! Fixed.